src/v/storage/disk_log_impl.cc

```diff
@@ -2335,7 +2335,8 @@ disk_log_impl::make_reader(timequery_config config) {
     vassert(!_closed, "make_reader on closed log - {}", *this);
     return _lock_mngr.range_lock(config).then(
       [this, cfg = config](std::unique_ptr<lock_manager::lease> lease) {
-          auto start_offset = _start_offset;
+          auto start_offset = cfg.min_offset;
```

Comment:
Wondering if it makes sense to `std::max(_start_offset,...)` here too, given the scheduling point from taking the lock.

Reply:
I don't think it matters since the range should only relevant segments anyway, regardless of start offset.

Reply:
I don't think we should, and this makes the code better in my opinion. The `cfg.min_offset` here acts as a  optimistic concurrency control mechanism. If the start offset did advance then our assumptions are not valid anymore. So we throw in the reader. The exception bubbles up to the client which, will retry the request.

If we clamp the start offset then the caller must be very careful to check whether the log advanced because data was moved to cloud or because there was a log truncation. And what if it forgets about doing that?

A bit opinionated and half-baked but I think down the road this will lead to better code.

Reply:
Yeah I'm fine with this. As a reader using the config as is seems less surprising